lib/elixir/lib/task.ex

```diff
@@ -338,13 +341,17 @@ defmodule Task do
   end
 
   @doc """
-  Receives a group of tasks and a message and finds
+  Takes a list of tasks and a message and finds
   a task that matches the given message.
 
-  This function returns a tuple with the returned value
-  in case the message matches a task that exited with
-  success alongside the matching task. It returns `nil`
-  if no task was found. It exits if the task has failed.
+  If the message matches a task that finished normally,
+  this function returns a tuple with the task's
+  result and `Task` struct.
+
+  If the message matches a task that terminated before
+  sending its result back, this function will exit.
```

Comment:
> Gone in master.

From 1.2 we can pattern match on maps with variable keys. Using a map of the form `%{reference() => any()}` will be much more efficient and likely cleaner than using `Task.find`. Particularly when using `async` because do not need to handle `:DOWN`s from tasks (unless trapping exits or calling `exit(:normal)` in the task).

> I don't think @fishcakez would be happy about people sticking try/rescue in their tasks, but, in my opinion, this would serve as a workaround approximating the desired behaviour of Task.find I described in the second paragraph of this comment.

I have suggested this approach on IRC a few times. When using a Task you should use exactly the same error handling as when not using a Task. Task is a method of providing concurrency, not error isolation.

I think this comes down to error handling in general. If a specific error is anticipated then handle (catch/rescue) it as close to the source as possible. If all errors are to be handled then the catch/trap point needs to be carefully thought out if its not going to be a supervisor.

> Normally, when someone is talking about a gen server spawning tasks, they imply receiving results from those tasks as messages and not killing the whole server because of one of the tasks terminating.

Is this true?